force-app/main/default/classes/SOQL.cls

```diff
@@ -107,19 +164,31 @@ public with sharing class SOQL {
      * @param keyword The keyword to search for in the Opportunity Name.
      * @return A list of Opportunity records.
      */
-    public static List<Opportunity> getOpportunitiesByNameKeyword(String keyword) {
-        return null; // Replace null with the variable you used to store the result
+   public static List<Opportunity> getOpportunitiesByNameKeyword(String keyword) {
+
+    if (keyword == null){
+        return new List<Opportunity>();
     }
 
-    /**
+    String searchWord = '%' + keyword + '%';
+
+    List<Opportunity> oppWithKeyword = [SELECT Id, Name, Amount FROM Opportunity WHERE Name LIKE :searchWord];
+   
+    return oppWithKeyword;
+   }
+
+  /** 
      * Question 9
      * Retrieves all Cases that Case Origin is not Web or Email or Blank
      * @return A list of Case records.
-     */
+    */ 
     public static List<Case> getFilterCaseOrigins() {
-        return null; // Replace null with the variable you used to store the result
-    }
 
+     List<Case> casesFiltered =   [SELECT Id,  Status, Subject, Origin FROM  Case WHERE Origin NOT IN ('Web', 'Email', NULL)];
```

Comment:
Try to use a list or set for the values.
Be advised that Null is not the same as blank